lib/iris/cube.py

```diff
@@ -2785,6 +2785,34 @@ def slices(self, ref_to_slice, ordered=True):
 
         return _SliceIterator(self, dims_index, dim_to_slice, ordered)
 
+    def invert_dims(self, dimensions):
+        """
+        Invert the specified dimensions of a cube in-place.
+
+        Args:
+
+        * dimensions (iterable of ints):
```

Comment:
> Since this affects the public API it should be ironed out now...

@ajdawson: Thanks for getting this back on my radar with the magic words "public API". :wink: 

@cpelley: :+1: for lazy tranpose, but `invert_dims` seems too niche to be on the Cube. What's stopping you from indexing your cube with a `::-1` slice? (NB. I'd prefer to deal with @ajdawson's excellent suggestion of being able to use coordinate names by using pandas/xarray style name-based indexing and have been trying to put together [a proposal](https://github.com/SciTools/iris/wiki/IEP-1) that would be relevant.)